File: expert-link-backend/src/main/java/com/expertlink/service/AuditLogService.java
package com.expertlink.service;

import com.expertlink.domain.AuditLog;
import com.expertlink.repository.AuditLogRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class AuditLogService {

    private final AuditLogRepository auditLogRepository;

    @Transactional
    public void record(String username, Long userId, String httpMethod, String requestUri, String action,
                       boolean success, String detail, String clientIp) {
        AuditLog row = AuditLog.builder()
                .username(username)
                .userId(userId)
                .httpMethod(httpMethod)
                .requestUri(truncate(requestUri, 500))
                .action(truncate(action, 300))
                .success(success)
                .detail(truncate(detail, 8000))
                .clientIp(truncate(clientIp, 64))
                .build();
        auditLogRepository.save(row);
    }

    @Transactional(readOnly = true)
    public Page<AuditLog> findAll(Pageable pageable) {
        return auditLogRepository.findAllByOrderByCreatedAtDesc(pageable);
    }

    private static String truncate(String s, int max) {
        if (s == null) {
            return null;
        }
        return s.length() <= max ? s : s.substring(0, max);
    }
}

File: expert-link-frontend/src/api/services/audit-log.service.ts
import { apiClient, handleApiError } from '../utils'
import type { ApiResponse, PaginatedResponse } from '../types'

export interface AuditLogRow {
  id: number
  username?: string
  userId?: number
  httpMethod?: string
  requestUri?: string
  action?: string
  success: boolean
  detail?: string
  clientIp?: string
  createdAt?: string
}

export class AuditLogService {
  static async list(page = 0, size = 20): Promise<PaginatedResponse<AuditLogRow>> {
    try {
      const response = await apiClient.get<ApiResponse<PaginatedResponse<AuditLogRow>>>('/audit-logs', {
        params: { page, size },
      })
      return response.data
    } catch (error) {
      throw handleApiError(error, '获取审计日志失败')
    }
  }
}
